src/McpServer.php

```
@@ -795,13 +798,19 @@ private function executeXDebug(string|int|null $id, array $args): JsonRpcRespons
             // Build command - user must specify PHP binary explicitly
             $cmd .= ' -- ' . $script;
 
-            // Execute command
-            $output = [];
+            // Execute command and redirect output to temp file (shutdown function output requires file redirect)
+            $tmpFile = tempnam(sys_get_temp_dir(), 'xstep_');
             $returnCode = 0;
-            exec($cmd . ' 2>&1', $output, $returnCode);
+            exec($cmd . ' > ' . escapeshellarg($tmpFile) . ' 2>&1', $execOutput, $returnCode);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$execOutput'. (undefined)

(UnusedLocalVariable)